src/shlPayload.ts

```diff
@@ -0,0 +1,146 @@
+import { HTTPError } from "got";
+import { ErrorCode } from "./error";
+import Log, { LogLevels } from "./logger";
+import { IOptions } from "./options";
+import { parseJson, post, unexpectedProperties } from "./utils";
+import * as shlManifest from "./shlManifest";
+
+export async function validate(shlinkPayloadJson: string, options: IOptions): Promise<Log> {
+    const log = new Log("SHL-Payload");
+
+    const payload = parseJson<ShlinkPayload>(shlinkPayloadJson);
+
+    if (!payload) {
+        return log.fatal(`Cannot decode payload as JSON`, ErrorCode.INVALID_SHLINK);
+    }
+
+    if (JSON.stringify(payload) !== shlinkPayloadJson) {
+        log.warn(
+            `JSON payload not minified (extraneous whitespace and/or formatting)`,
+            ErrorCode.SHLINK_VERIFICATION_ERROR
+        );
+    }
+
+    if (typeof payload?.key !== "string" || !/^[0-9a-zA-Z-_]{43}$/.test(payload.key)) {
```

Comment:
I think `Z-_` is treated as a character range rather than a set of 3 acceptable chars:

```js
/^[Z-_]{1}$/.test("Z")
true

/^[Z-_]{1}$/.test("_")
true

/^[Z-_]{1}$/.test("-")
false // !!
```

Note: this issue occurs in 5 other spots too

Reply:
`Z-_` is a range.  However, it groups the ranges from left to right so `[Z-_]` is a range, but 
`[A-Z-_]` is  `(A-Z|-|_)`

```javascript
/^[A-M-Z]{1}$/.test("S")
false

/^[A-M-Z]{1}$/.test("-")
true
```

Usually when doing a regex for base64url I use `[\w-]+` as `\w` is `[a-zA-Z0-9_]` 
I thought expanding it out would make it more clear to the next dev that came along - but then I made it look more confusing :)

Reply:
D'oh! Thanks for spelling this out for me :-)